deepchem/utils/evaluate.py

```diff
@@ -287,6 +287,17 @@ def compute_model_performance(
       If `per_task_metrics == True`, then returns a second dictionary
       of scores for each task separately.
     """
+    if n_classes is not None:
+      logger.warning(
+          "n_classes is deprecated as an argument and will be removed in a future version of DeepChem."
+          "In future version, n_classes will be directly inferred from the model"
+      )
+    if hasattr(self.model, 'n_classes'):
+      if n_classes != self.model.n_classes:
```

Comment:
I think you want this to be inside the `if n_classes is not None` block?  Otherwise you'll get a warning any time the model specifies `n_classes`.

Reply:
Thanks for noticing it, I missed it. I have updated it.